src/text-expander-element.ts

```diff
@@ -86,16 +86,21 @@ class TextExpander {
     this.combobox.navigate(1)
   }
 
-  deactivate() {
+  deactivate(cursor?: number) {
```

Comment:
How tightly connected is the cursor with deactivation logic? I have concerns if it should be part of this function. For example in `onCommit()` we are setting `this.lookBackIndex` again on line 133. In `onKeydown()` we are passing both `this.input.selectionEnd || this.lookBackIndex` while `this.lookBackIndex` is used in the body of that function. Did you consider having separate function for dealing with cursor updates?

Reply:
I did consider adding it separate, i think there was a reason i had to pass it here. But good point, i'll try to de-couple it and set the lookbackIndex separately in each function

Reply:
Yeah, if there was some edge case I would not insist on the refactor, I suggest though to add a test checking it or at least a comment that will give some context about this edge case.

Reply:
I changed it and could find any edge case. I hope i am not missing something

Reply:
Ah, i think the problem will arise when dismiss will be triggered from outside of the webComponent (like https://github.com/github/github/pull/163808) . Yes that was the issue. it needs to be set to the current cursor position when it is triggered from outside the component

Reply:
Good point. It could be controlled though inside `dismiss()` implementation I guess.